src/OpenSpaceToolkit/Astrodynamics/Estimator/TLESolver.cpp

```diff
@@ -0,0 +1,307 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Utility.hpp>
+
+#include <OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Estimator/TLESolver.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/BrouwerLyddaneMean/BrouwerLyddaneMeanLong.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace estimator
+{
+
+using ostk::core::type::Shared;
+
+using ostk::mathematics::object::VectorXd;
+
+using ostk::physics::coordinate::Frame;
+using ostk::physics::coordinate::Position;
+using ostk::physics::coordinate::Velocity;
+using ostk::physics::environment::gravitational::Earth;
+using ostk::physics::time::Instant;
+
+using ostk::astrodynamics::trajectory::orbit::model::blm::BrouwerLyddaneMeanLong;
+using ostk::astrodynamics::trajectory::orbit::model::SGP4;
+
+TLESolver::Analysis::Analysis(const TLE& aDeterminedTLE, const LeastSquaresSolver::Analysis& anAnalysis)
+    : determinedTLE(aDeterminedTLE),
+      solverAnalysis(anAnalysis)
+{
+}
+
+std::ostream& operator<<(std::ostream& anOutputStream, const TLESolver::Analysis& anAnalysis)
+{
+    anAnalysis.print(anOutputStream);
+
+    return anOutputStream;
+}
+
+void TLESolver::Analysis::print(std::ostream& anOutputStream) const
+{
+    ostk::core::utils::Print::Header(anOutputStream, "Analysis");
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Determined TLE");
+    ostk::core::utils::Print::Line(anOutputStream) << determinedTLE;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Analysis");
+    solverAnalysis.print(anOutputStream);
+
+    ostk::core::utils::Print::Footer(anOutputStream);
+}
+
+TLESolver::TLESolver(
+    const LeastSquaresSolver& aSolver,
+    const Integer& aSatelliteNumber,
+    const String& anInternationalDesignator,
+    const Integer& aRevolutionNumber,
+    const bool aFitWithBStar,
+    const Shared<const Frame>& anEstimationFrameSPtr
+)
+    : solver_(aSolver),
+      satelliteNumber_(aSatelliteNumber),
+      internationalDesignator_(anInternationalDesignator),
+      revolutionNumber_(aRevolutionNumber),
+      fitWithBStar_(aFitWithBStar),
+      estimationFrameSPtr_(anEstimationFrameSPtr),
+      defaultBStar_(1e-4),
+      firstDerivativeMeanMotionDividedBy2_(0.0),
+      secondDerivativeMeanMotionDividedBy6_(0.0),
+      ephemerisType_(0),
+      elementSetNumber_(0),
+      tleStateBuilder_(StateBuilder::Undefined())
+{
+    // Setup coordinate subsets for TLE state
+    Array<Shared<const CoordinateSubset>> coordinateSubsets = {
+        std::make_shared<CoordinateSubset>("INCLINATION", 1),
```

Comment:
tbh I think we should add these as actually coordinate subset classes

Reply:
Yeah we can add some static constructors. Not a priority IMO since it's only used internally in the solver.